proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -0,0 +1,221 @@
+
+There are certain resource kinds within the SRV resource
+class that are "sampleable". These resources are typically
+prefixed with "Texture" in actual HLSL source. So, there will
+be a special check to see whether the candidate resource is 
+prefixed with "Texture", and if so, the "s" register type will
+also be permitted for that resource.
+
+If the given candidate resource is a user-defined type (UDT), then further
+analysis is necessary. The first step is to gather all registers that
+are being bound to this declaration, and collect the register types
+that are being specified. The UDT must have at least one valid resource
+that can be bound to the provided register type. If not, an error must be
+emitted stating that "No resource contained in struct '%0' can be bound
+to register type '%1'". There are no issues if a UDT has more resources
+than there are register binding statements, the resources will be bound to
+the next available space automatically, and so compilation can succeed.
+Additionally, a singular resource can be bound multiple times to different
+spaces, so multiple register types are allowed for a single resource.
+Below is a table that represents the different behaviors that could arise
+given some examples of different UDT's:
+
+| Code | Diagnostic |
+|-|-|-|
+| struct Foo {<br>  float f;<br>  Buffer<float> Buf;<br>  RWBuffer<float> RWBuf;<br>};<br> Foo f : register(t0) : register(s0); | None. Success because Buf gets bound to t0 and RWBuf gets bound to s0, and f is skipped. 
+| struct Foo {<br>  float f;<br>  Buffer<float> Buf;<br>  RWBuffer<float> RWBuf;<br>  RWBuffer<float> RWBuf2;<br>};<br> Foo f : register(t0) : register(s0); | None. Success because Buf gets bound to t0  and RWBuf gets bound to s0, and f is skipped. RWBuf2 gets assigned to s1 even though there is no explicit binding for s1. |
+| struct Foo {<br>  float f;<br>  Buffer<float> Buf;<br>}; <br> Foo f : register(t0) : register(s0); | None. Success because Buf gets bound to t0. Buf will also be bound to s0.|
+| struct Foo {<br>  struct Bar {<br>    RWBuffer<int> a;<br>    };<br>    Bar b;<br>};<br> Foo f : register(t0) | None. Success because Bar, the struct within Foo, has a valid resource that can be bound to t0. |
+| struct Foo {<br>  float f;<br>}; <br> Foo f : register(t0) | DefaultError warning. "UDT resource 'Foo' does not contain an applicable resource type for register type 't'"|
+| struct Foo {<br>  struct Bar {<br>      float f;<br>    }<br>    Bar b;<br>};<br> Foo f : register(t0) | DefaultError warning. "UDT resource 'Foo' does not contain an applicable resource type for register type 't'"|
+
+Finally, if the candidate type is not a valid resource type or not a UDT,
+the final case will be entered. Types that can be immediately determined
+to not be a resource type or not contain a resource type (types like booleans,
+int, float, float4, etc) are known as "non-intangible". This is because

```

Comment:
Ah, ok.  I thought maybe "non-intangible" was a mistake.  To avoid the double-negative, how about calling them "tangible"?